course7/homeworks/patrickkostjens/complex/CMakeLists.txt

```diff
@@ -6,9 +6,9 @@ find_package(Boost REQUIRED COMPONENTS filesystem)
 
 INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
 
-add_library(print SHARED print.cc complex.cc complex.hh)
+add_library(print SHARED print.cc complex.cc complex.hh complex_stack.cc complex_stack.hh)
```

Comment:
You could create a small static library with complex and stack, and then link both the executable and the plugin with it.
This would be the interface library you'd distribute to plugin writers.